pallets/liquidity-pools-gateway/src/message_processing.rs

```diff
@@ -14,24 +14,21 @@ use sp_arithmetic::traits::{EnsureAddAssign, EnsureSub};
 use sp_runtime::DispatchError;
 
 use crate::{
-	message::GatewayMessage, Config, Error, InvalidMessageSessions, Pallet, PendingInboundEntries,
-	Routers, SessionIdStore,
+	message::GatewayMessage, Config, Error, Pallet, PendingInboundEntries, Routers, SessionIdStore,
 };
 
-/// The limit used when clearing the `PendingInboundEntries` for invalid
-/// session IDs.
-const INVALID_ID_REMOVAL_LIMIT: u32 = 100;
-
 /// Type used when storing inbound message information.
 #[derive(Debug, Encode, Decode, Clone, Eq, MaxEncodedLen, PartialEq, TypeInfo)]
 #[scale_info(skip_type_params(T))]
 pub enum InboundEntry<T: Config> {
 	Message {
+		session_id: T::SessionId,
```

Comment:
Instead of `Map<Proof, T::RouterId, (T::SessionId, InboundEntry)>`, I added the session entry to the actual entry. This gives us a cleaner storage layout and it's definitely easier to deal with it given all the `try_mutate` calls.

Lemme know what you think @lemunozm

Reply:
Awesome! yeah, that was the intention 🙌🏻